packages/caver-klay/caver-klay-accounts/src/index.js

```diff
@@ -875,6 +892,72 @@ Accounts.prototype.signTransactionWithSignature = function signTransactionWithSi
     });
 };
 
+/**
+ * combineSignatures combines RLP encoded raw transaction strings.
+ * combineSignatures compares transaction before combining, and if values in field are not same, this throws error.
+ * The comparison allows that the address of the fee payer is '0x'(default value) for some transactions while the other transactions have a specific fee payer. This is for the use case that some transactions do not have the fee payer's information.
+ * In this case, feePayer field doesn't have to be compared with other transaction.
+ * 
+ * @method combineSignatures
+ * @param {Array} rawTransactions The array of raw transaction string to combine.
+ * @return {Object}
+ */
+Accounts.prototype.combineSignatures = function combineSignatures(rawTransactions) {
+  let decodedTx
+  let senders = []
+  let feePayers = []
+  let feePayer
+
+  if (!_.isArray(rawTransactions)) throw new Error(`The parameter of the combineSignatures function must be an array of RLP encoded transaction strings.`)
+  
+  for (const raw of rawTransactions) {
+    const { senderSignatures, feePayerSignatures, decodedTransaction } = extractSignatures(raw)
+
+    senders = senders.concat(senderSignatures)
+    feePayers = feePayers.concat(feePayerSignatures)
+    
+    if (decodedTx) {
+      let isSame = true
+      const keys = Object.keys(decodedTx)
+      for (const key of keys) {
+        if (key === 'v' || key === 'r' || key === 's' || key === 'signatures' || key === 'payerV' || key === 'payerR' || key === 'payerS' || key === 'feePayerSignatures') {
+          continue
+        }
+
+        // feePayer field can be '0x' when after sender signs to trasnaction.
+        // For handling this, if feePayer is '0x', don't compare with other transaction
+        if (key === 'feePayer') {
+          if (decodedTx[key] === '0x') {
+            decodedTx[key] = decodedTransaction[key]
+            feePayer = decodedTx[key]
+          } else if (decodedTransaction[key] === '0x') {
+            continue
+          }
+        }
+
+        if (decodedTransaction[key] === undefined || decodedTx[key] !== decodedTransaction[key]) {
+          isSame = false
+          break
+        }
+      }
+      if (!isSame) throw new Error(`Failed to combineSignatures: Signatures that sign to different transaction cannot be combined.`)
+    } else {
+      decodedTx = decodedTransaction
+    }
+  }
+
+
+  const parsedTxObject = decodeFromRawTransaction(rawTransactions[0])
+  if (feePayer) parsedTxObject.feePayer = feePayer
+  
+  parsedTxObject.signatures = senders
+  if (parsedTxObject.feePayer && parsedTxObject.feePayer !== '0x' && parsedTxObject.feePayerSignatures) {
```

Comment:
Do we need to check `parsedTxObject.feePayerSignatures`?
If feePayer is not set in the first transaction, I think it has possibility that feePayerSignatures is not set.

Reply:
'feePayers' variable gather all feePayerSignatures in rawTransaction strings.(L917)
So, this line check, fee delegated transaction or not and condition which determine can include feePayerSignatures through check address of feePayer.
At L951, feePayer addess is set, so at L954, feePayer should not '0x' for including feePayerSignatures.
At L954 checking parsedTxObject.feePayerSignatures is for confirming result of decodeTransaction has feePayerSignatures field.

Reply:
Dont we think about the case that feePayerSignature is not set for the firt transaction? This implementation assumes that the first transaction does not have feePayer field.

Reply:
feePayers contains all the feePayerSignatures in the rawTransaction. So I don't think feePayerSignatures is empty in the decoded result of the first transaction.

Seems to check for conditions that don't require a bit of logic, so I changed the logic a bit in the added commit.